src/helpers.js

```diff
@@ -48,12 +48,22 @@ async function getAssets(folder) {
   assets.push({
     ...indexHTML,
     path: '/',
+    name: '/',
   });
   assets.push({
     ...indexHTML,
     path: '/login',
+    name: '/login',
   });
 
+  // Add Asset SIDs to ensure that current assets are updated
+  if (appInfo && appInfo.hasAssets) {
+    for (const newAsset of assets) {
+      const currentAsset = appInfo.assets.find(currentAsset => currentAsset.friendlyName === newAsset.name);
+      newAsset.sid = currentAsset.sid;
```

Comment:
I don't think that does anything when you pass it to `deployProject`. What's the scenario here?
In theory this is what `getOrCreateAssetResource` already does when you call `deployProject`:

https://github.com/twilio-labs/serverless-api/blob/master/src/api/assets.ts#L78

Reply:
The scenario is that a developer is trying to use the `--override` flag when `--app-directory` is also specified. Setting the asset SID allows the deploy to update the existing asset with the current file being deployed. Not specifying the asset SID results in the twilio CLI failing stating that an "existing asset has a conflicting sid"

Reply:
When there are new local assets to upload this section throws an error.  When an asset is new then `currentAsset` is undefined.

```
      if (currentAsset) {
        newAsset.sid = currentAsset.sid;
      }
```
Also, the there are unit tests for `getAssets` that should be updated.  Let me know if I should do that.

Reply:
I see. So this whole section is actually not needed:
```
  if (appInfo && appInfo.hasAssets) {
    for (const newAsset of assets) {
      const currentAsset = appInfo.assets.find(currentAsset => currentAsset.friendlyName === newAsset.name);
      newAsset.sid = currentAsset.sid;
    }
  }
```
As @dkundel has pointed out, this is already being handled by `deployProject`. 

It used to throw the `existing asset has a conflicting sid` error, but I think that was fixed when you added the `name` property on lines 51 and 56.

Reply:
Yeah ideally this should be handled by the `name` property.

Reply:
Awesome thanks y'all!